### PR TITLE
shard coordinator should wait until min-members regions registered, #21194

### PR DESCRIPTION
**Pull request by patriknw**
Refs #21194

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM (fail is mima)

**Comment by patriknw** [comment]
fixed mima, was a rename of an internal actor method

**Comment by patriknw** [comment]
I have rebased this (just mima) but I would like another lgtm before pushing and merging to avoid overloading jenkins

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala [hunk 1]
```diff
@@ -460,7 +465,11 @@ abstract class ShardCoordinator(typeName: String, settings: ClusterShardingSetti
       }
 
     case GetShardHome(shard) ⇒
-      if (!rebalanceInProgress.contains(shard)) {
+      if (rebalanceInProgress.contains(shard)) {
+        log.debug("GetShardHome [{}] request ignored, because rebalance is in progress for this shard.", shard)
+      } else if (!hasAllRegionsRegistered()) {
+        log.debug("GetShardHome [{}] request ignored, because all regions have not registered yet.", shard)
```
Comment (drewhk) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: "not all regions have registered yet"

**Review by drewhk — APPROVED** [review]
LGTM

File: akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala (hunk 1 above)
Comment (patriknw) [inline akka-cluster-sharding/src/main/scala/akka/cluster/sharding/ShardCoordinator.scala]: thx, I'll update that

**Pull request merged**